Orange/widgets/unsupervised/owpca.py

```diff
@@ -375,6 +386,26 @@ def _update_normalize(self):
         if self.data is None:
             self._invalidate_selection()
 
+    def _init_projector(self):
+        if self.center:
+            self._pca_projector = PCA(max_components=MAX_COMPONENTS)
+            self._pca_projector.component = self.ncomponents
+            self._pca_preprocessors = PCA.preprocessors
+        else:
+            # Equivalent to PCA, but without data centering
+            self._pca_projector = TruncatedSVD(max_components=MAX_COMPONENTS)
+            self._pca_projector.component = self.ncomponents
+            self._pca_preprocessors = TruncatedSVD.preprocessors
+
+    def _update_center(self):
+        if self.center and self.data is not None and self.data.is_sparse():
```

Comment:
This code is currently unreachable, if I am not mistaken. You are setting `self.center = False` if data is sparse in `set_data`. It is here as a sanity check?